core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/recoverer.go

```diff
@@ -596,3 +619,23 @@ func (r *logRecoverer) removePending(workID string) {
 	}
 	r.pending = updated
 }
+
+// sortPending sorts the pending list by a random order based on the normalized latest block number.
+// Divided by 10 to ensure that nodes with similar block numbers won't end up with different order.
+// NOTE: the lock must be held before calling this function.
+func (r *logRecoverer) sortPending(latestBlock uint64) {
+	normalized := latestBlock / 10
```

Comment:
nit. 10 might not be enough on fast chains like arb. 100 might be better (but then its slow on other chains also )

Reply:
We know the block time within the recoverer, we could also compute something based on the block time.

Reply:
that can have further issues when blockTime is not synchronized 😅